src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -238,62 +211,58 @@ private static boolean shouldBeOnSameLine(RightCurlyOption bracePolicy, Details
      * Whether right curly should be alone on line when ALONE option is used.
      * @param details details for validation.
+     * @param targetSrcLine A string with contents of rcurly's line
      * @return true, if right curly should be alone on line when ALONE option is used.
      */
-    private static boolean shouldBeAloneOnLineWithAloneOption(Details details) {
-        return !isAloneOnLine(details)
+    private static boolean shouldBeAloneOnLineWithAloneOption(Details details,
+                                                              String targetSrcLine) {
+        return !isAloneOnLine(details, targetSrcLine)
                 && !isEmptyBody(details.lcurly);
     }
 
     /**
      * Whether right curly should be alone on line when ALONE_OR_SINGLELINE option is used.
      * @param details details for validation.
+     * @param targetSrcLine A string with contents of rcurly's line
      * @return true, if right curly should be alone on line
      *         when ALONE_OR_SINGLELINE option is used.
      */
-    private static boolean shouldBeAloneOnLineWithAloneOrSinglelineOption(Details details) {
-        return !isAloneOnLine(details)
+    private static boolean shouldBeAloneOnLineWithAloneOrSinglelineOption(Details details,
+                                                                          String targetSrcLine) {
+        return !isAloneOnLine(details, targetSrcLine)
                 && !isSingleLineBlock(details)
-                && !isAnonInnerClassInit(details.lcurly)
+                && !isAnonInnerClassInitCorrect(details)
                 && !isEmptyBody(details.lcurly);
     }
 
     /**
-     * Whether right curly brace starts target source line.
-     * @param details Details of right curly brace for validation
-     * @param targetSourceLine source line to check
-     * @return true if right curly brace starts target source line.
-     */
-    private static boolean isOnStartOfLine(Details details, String targetSourceLine) {
-        return CommonUtils.hasWhitespaceBefore(details.rcurly.getColumnNo(), targetSourceLine)
-                || details.lcurly.getLineNo() == details.rcurly.getLineNo();
-    }
-
-    /**
      * Checks whether right curly is alone on a line.
      * @param details for validation.
+     * @param targetSrcLine A string with contents of rcurly's line
      * @return true if right curly is alone on a line.
      */
-    private static boolean isAloneOnLine(Details details) {
+    private static boolean isAloneOnLine(Details details, String targetSrcLine) {
         final DetailAST rcurly = details.rcurly;
-        final DetailAST lcurly = details.lcurly;
         final DetailAST nextToken = details.nextToken;
-        return rcurly.getLineNo() != lcurly.getLineNo()
-            && rcurly.getLineNo() != nextToken.getLineNo();
+        return rcurly.getLineNo() != nextToken.getLineNo()
+                && CommonUtils.hasWhitespaceBefore(details.rcurly.getColumnNo(), targetSrcLine);
```

Comment:
yes ! the methods were previously static so I let them be ! that's the only reason why `targetSrcLine` is being passed aroung ! Should I make them non-static so that the `getLine` method be called from them ?